### PR TITLE
Feature/dockerfile step

### PR DESCRIPTION
**Pull request by Jyuart**
User now can switch to "Dockerfile mode" while creating build steps.
image
He will be prompted to agree deleting all his build steps.
In this mode he have to specify location of Dockerfile in his repository. He also can't add build steps. There is special button for switching back to custom build steps.
image

### CODE REVIEW

File: backend/buildeR.Processor/Services/ProcessorService.cs [hunk 1]
```diff
@@ -93,9 +94,19 @@ public async Task BuildProjectAsync(ExecutiveBuildDTO build)
             CloneRepository(build.RepositoryUrl, pathToClonedRepository, build.BranchName);
 
             try
-            {
-                var dockerFileContent = GenerateDockerFileContent(build.BuildSteps, build.RepositoryUrl);
-                await CreateDockerFileAsync(dockerFileContent, pathToClonedRepository);
+            
+                {
+                if (build.BuildSteps.Count() != 0 && build.BuildSteps.FirstOrDefault().BuildStepName == "Dockerfile")
```
Comment (VoBilyk) [inline backend/buildeR.Processor/Services/ProcessorService.cs:99]: 1. "build.BuildSteps.Count() != 0" => "build.BuildSteps.Any()"
2. FirstOrDefault().BuildStepName will throw exception if it's default

File: frontend/src/app/modules/project/project-build-steps/project-build-steps.component.ts [hunk 2]
```diff
@@ -102,6 +145,9 @@ export class ProjectBuildStepsComponent extends BaseComponent implements OnInit,
           this.allBuildSteps = [];
           this.allBuildSteps = this.allBuildSteps.concat(this.buildSteps);
           this.isLoading = false;
+          if (this.allBuildSteps.length !== 0 && this.allBuildSteps[0].buildStepName === 'Dockerfile') {
```
Comment (VoBilyk) [inline frontend/src/app/modules/project/project-build-steps/project-build-steps.component.ts:148]: 1. "this.allBuildSteps.length !== 0" => this.allBuildSteps.length

**Review by VoBilyk — APPROVED** [review]

**Comment by VoBilyk** [comment]
Try to simplify some code constructions

**Comment by Jyuart** [comment]
> Try to simplify some code constructions

Ok, I'll work on this.

**Pull request merged**